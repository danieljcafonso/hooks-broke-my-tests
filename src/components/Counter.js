import React, { useState } from "react";


const Counter = () => {
	const [counter, setCounter] = useState(0)

	return (
		<div>
			<h2>{counter}</h2>
			<button className="incrementButton" onClick={() => setCounter((count) => count + 1) }> +1 </button>
			<button className="decrementButton" onClick={() => setCounter((count) => count - 1) }> -1 </button>
		</div>
	);
}
export default Counter;