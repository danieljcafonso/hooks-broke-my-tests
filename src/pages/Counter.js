import React from "react";
import CounterComponent from '../components/Counter'
const Counter = () => {
	return <CounterComponent />;
};
export default Counter;
