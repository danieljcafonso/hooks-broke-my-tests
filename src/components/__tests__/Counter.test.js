import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Counter from "../Counter";

describe("Counter", () => {

	test("should render Counter", () => {
		const { container } = render(<Counter />);
		expect(container).toMatchSnapshot();
	});

	test("should have default value", () => {
		const { container } = render(<Counter />);
		const text = container.firstChild;
		expect(text).toHaveTextContent("0");
	});

	test("incrementButton should have default value", () => {
		const { getByText } = render(<Counter />);
		expect(getByText("+1")).toBeInTheDocument();
	});

	test("decrementButton should have default value", () => {
		const { getByText } = render(<Counter />);
		expect(getByText("-1")).toBeInTheDocument();
	});

	test("should call increment button onClick ", () => {
		const { getByText, container } = render(<Counter />);
		const text = container.firstChild;
		fireEvent.click(getByText("+1"));
		expect(text).toHaveTextContent("1");
	});

	test("should call decrement button onClick ", () => {
		const { getByText, container } = render(<Counter />);
		const text = container.firstChild;
		fireEvent.click(getByText("-1"));
		expect(text).toHaveTextContent("-1");
	});
});
