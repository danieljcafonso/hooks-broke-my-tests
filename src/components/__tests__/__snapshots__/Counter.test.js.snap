// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render Counter 1`] = `
<div>
  <div>
    <h2>
      0
    </h2>
    <button
      class="incrementButton"
    >
       +1 
    </button>
    <button
      class="decrementButton"
    >
       -1 
    </button>
  </div>
</div>
`;
